### PR TITLE
fix(metrics): collect node num metrics at process level

### LINKED ISSUE #5186: improve `node_num` metrics
image

Today I saw a `-1` node number. That implies that maintaining the node_num metric with `dec()`/`inc()` is not reliable.

I think it's better to report (a snapshot of) the current number of working to Prometheus in some fixed interval, for example, when collecting `process_resident_memory_bytes`, instead of incresing/decresing it when changes happen. 

_Originally posted by @fuyufjh in https://github.com/risingwavelabs/risingwave/issues/4933#issuecomment-1239196021_

### PR DESCRIPTION
**Pull request by Gun9niR**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

RisingWave dashboard already reports all the workers from meta node. So for grafana, the node num will be updated once the process starts.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)

Closes #5186

### CODE REVIEW

File: src/common/src/monitor/node.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+struct NodeCollector {
+    descs: Vec<Desc>,
+    // Always set to `1`.
+    node_num: IntCounter,
+}
+
+impl NodeCollector {
+    pub fn new() -> Self {
+        let node_num = IntCounter::with_opts(Opts::new("node_num", "The number of node.")).unwrap();
```
Comment (fuyufjh) [inline src/common/src/monitor/node.rs:36]: Should it be a Gauge instead of Counter?

Comment (fuyufjh) [reply src/common/src/monitor/node.rs:36]: https://prometheus.io/docs/concepts/metric_types/#counter

> A counter is a cumulative metric that represents a single [monotonically increasing counter](https://en.wikipedia.org/wiki/Monotonic_function) whose value can only increase or be reset to zero on restart

Comment (Gun9niR) [reply src/common/src/monitor/node.rs:36]: We don't need to decrease the value. When a process is killed, its corresponding metric will be gone, resulting in a decrease in the sum.

**Review by fuyufjh — APPROVED** [review]

**Comment by fuyufjh** [comment]
`grafana/risingwave-dashboard.json` conflicts again 😅

**Comment by fuyufjh** [comment]
@Gun9niR New problem:

image

Once there is no compute nodes alive, the graph shows "missing data" instead of 0.

**Pull request merged**